### PR TITLE
fix(a11y): vertical nav-add label to expand/collapse button

### PR DESCRIPTION
**Pull request by andyfeds**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: [CDE-1391](https://jira.eng.vmware.com/browse/CDE-1391)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by sandeepkulki — APPROVED** [review]

**Review by sandeepkulki — APPROVED** [review]

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]
This change needs unit tests.

File: projects/angular/src/layout/vertical-nav/vertical-nav-group.spec.ts [hunk 1]
```diff
@@ -232,6 +232,13 @@ export default function (): void {
         fixture.detectChanges();
         expect(button.getAttribute('aria-expanded')).toBe('true');
       });
+
+      it('should set the aria-label attribute on the expand/collapse button', () => {
+        const button: HTMLButtonElement = compiled.querySelector('.nav-group-trigger');
+        const navText: HTMLSpanElement = compiled.querySelector('.nav-text');
+        expect(button.hasAttribute('aria-label')).toBe(true);
+        expect(button.getAttribute('aria-label')).toContain(navText.textContent.trim());
```
Comment (kevinbuhmann) [inline projects/angular/src/layout/vertical-nav/vertical-nav-group.spec.ts:240]: The expected value should be a literal value, not a computer value.

File: projects/angular/src/layout/vertical-nav/vertical-nav-group.spec.ts [hunk 2]
```diff
@@ -232,6 +232,12 @@ export default function (): void {
         fixture.detectChanges();
         expect(button.getAttribute('aria-expanded')).toBe('true');
       });
+
+      it('should set the aria-label attribute on the expand/collapse button', () => {
+        const button: HTMLButtonElement = compiled.querySelector('.nav-group-trigger');
+        expect(button.hasAttribute('aria-label')).toBe(true);
+        expect(button.getAttribute('aria-label')).toContain('Link');
```
Comment (kevinbuhmann) [inline projects/angular/src/layout/vertical-nav/vertical-nav-group.spec.ts:239]: This is not correct.
1. The button has visible text that says "Group". Why do we need an aria-label on a button that has visible text?
2. There is a link in the group with "Link" as the visible text (and accessible name). Accessible names must be unique.

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]
See my comment. This change breaks the component by introducing incorrect and duplicate accessible names.

I think the referenced bug (CDE-1391) is a bug in the vertical nav routing story, not the vertical nav component. The top-level entries should NOT be links. That usage is incorrect and breaks the compoennt.

File: projects/angular/src/layout/vertical-nav/vertical-nav-group.spec.ts (hunk 2 above)
Comment (andyfeds) [reply projects/angular/src/layout/vertical-nav/vertical-nav-group.spec.ts:239]: You're right, It seems that there is an issue with how the component is implemented in storybook and not the component itself.  There is a related JIRA to track this: [CDE-1390](https://jira.eng.vmware.com/browse/CDE-1390)
Closing this PR.

**Pull request closed without merging**